### PR TITLE
Allow to create attribute values from product form

### PR DESCRIPTION
**Pull request by elwoodxblues**
This PR is follow-up to #1964. Just as with variant attributes, it introduces the ability to create new attribute values from the product form. 

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Python code quality checks pass: `pycodestyle`, `pydocstyle`, `pylint`.
1. [x] JavaScript code quality checks pass: `eslint`.

### CODE REVIEW

File: saleor/dashboard/product/forms.py [hunk 1]
```diff
@@ -209,10 +209,16 @@ def save(self, commit=True):
...
         for attr in self.product_attributes:
             value = self.cleaned_data.pop(attr.get_formfield_name())
-            if isinstance(value, AttributeChoiceValue):
-                attributes[smart_text(attr.pk)] = smart_text(value.pk)
-            else:
-                attributes[smart_text(attr.pk)] = value
+
+            # if the passed attribute value is a string,
+            # create the attribute value.
+            if not isinstance(value, AttributeChoiceValue):
```
Comment (akjanik) [inline saleor/dashboard/product/forms.py]: This block is the same as in `VariantAttributeForm`, maybe we can make a function from it?

**Comment by elwoodxblues** [comment]
@akjanik You were right about duplicated code, so I've extracted a form mixin that handles dynamic attributes fields in the same way in both product and variant form. I think the code looks a bit better now.

Also, I've taken the opportunity and dropped `get_attribute` and `set_attribute` functions from models, that were used mostly in tests. There is definitely too many confusing utility functions related to attributes, that we should eventually organize.

File: saleor/dashboard/product/forms.py [hunk 2]
```diff
@@ -137,7 +137,53 @@ def clean(self):
...
+            if attribute.has_values():
+                field = ModelChoiceOrCreationField(
+                    queryset=attribute.values.all(), **field_defaults)
+            else:
+                field = forms.CharField(**field_defaults)
+            self.fields[attribute.get_formfield_name()] = field
+
+    def iter_attribute_fields(self):
+        for attr in self.available_attributes:
+            yield self[attr.get_formfield_name()]
```
Comment (mad-anne) [inline saleor/dashboard/product/forms.py]: Shouldn't it be rather `self[attr.get_formfield_name()]`?

Comment (elwoodxblues) [reply saleor/dashboard/product/forms.py]: You can access fields in this way as well. It was here before so I didn't change it.

**Pull request merged**